**OBLIGATORIO FINAL G448 - Data Wrangling**
<br>**INTEGRANTES:**
<br>
+ Theo Pereira, 5703834-3, theopereira2001@gmail.com
+ Carlos Píriz, 4298784-4, piriz.carlos@gmail.com
+ Gabriel	Better,3138871-2, gabrielbetter@gmail.com

In [1]:
### Importacion de librerias ###

# Sistema
import os

# Tratamiento de datos
import pandas as pd
import numpy as np
import datetime as dt
import math

# Visualizacion 
import seaborn as sns
import matplotlib as mpl
from matplotlib import ticker
from matplotlib.ticker import FuncFormatter
from babel.numbers import format_currency
import matplotlib.pyplot as plt
%matplotlib inline

**CONSIGNAS**
<br>
El objetivo general es realizar un análisis exploratorio de los anuncios inmobiliarios de
algunas provincias de la Patagonia Argentina reportados por el portal Airbnb.
<br>
Los datos están conformados por cuatro archivos csv:
+ Neuquen.csv
+ Rio_Negro.csv
+ Chubut.csv
+ Tierra_del_Fuego.csv

**<h1>1. Carga de datos y armado del dataset</h1>**

In [2]:
# Carga de archivos de datos de chubut
df_chubut = pd.read_csv('https://raw.githubusercontent.com/carlos1680/CPEDWObligatorio/main/Chubut.csv')
df_chubut["provincia"]="CHUBUT"
df_chubut.sample(10)

id_alojamiento                                               name  \
90             37836757                                     Rincón del Sol   
208            21205321                                MI CASA... TU CASA.   
193  839910728991938719                       "El Rancho" Sentirte en casa   
240  894918223105948004                                  hospedaje Lorenzo   
216            53336620               Comodo Monoambiente en puerto madryn   
37   658010538920592684  Hermoso depto en Comodoro Rivadavia de 3 ambie...   
146  846474870173281355                       Contenedor a orillas del Río   
170            28221139                                           866 Casa   
168             4854900                                   Vistahue soñado!   
255  803666373797660065                      Casa nueva equipada con patio   

         category     rating                city   latitude  longitude  \
90    entire_home    4,4 (5)       Puerto Madryn -42.751650 -65.053520   
208  private_room   5,0 (18)    Puerto Piramides -42.569600 -64.277880   
193   entire_home        NaN          Lago Puelo -42.034017 -71.545491   
240   entire_home      Nuevo  Comodoro Rivadavia -45.859679 -67.496781   
216   entire_home   5,0 (16)       Puerto Madryn -42.767020 -65.037330   
37    entire_home   4,7 (33)  Comodoro Rivadavia -45.860380 -67.486250   
146   entire_home        NaN              Trelew -43.311848 -65.373233   
170   entire_home  4,89 (53)              Esquel -42.914820 -71.326320   
168   entire_home   5,0 (11)            Cushamen -42.527040 -71.529130   
255   entire_home    5,0 (5)       Puerto Madryn -42.787870 -65.024040   

                                       characteristics    check_in  \
90   ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-01   
208  ['3 camas', '2 dormitorios', '1,5 baños compar...  2023-08-17   
193              ['3 camas', '1 dormitorio', '1 baño']  2023-07-01   
240              ['2 camas', '1 dormitorio', '1 baño']  2023-07-01   
216  ['1 cama', 'Monoambiente', '1 baño', '1 cama q...  2023-07-03   
37   ['2 camas', '2 dormitorios', '1 ,5 baños', 'Ca...  2023-07-01   
146  ['No hay evaluaciones', '1 cama', '1 dormitori...  2023-07-25   
170  ['4 camas', '2 dormitorios', '1 baño', '1 cama...  2023-07-01   
168  ['8 camas', '4 dormitorios', '3,5 baños', 'Cal...  2023-07-01   
255  ['4 camas', '1 dormitorio', '1 baño', 'Calific...  2023-07-02   

      check_out price_discounted price_original price_qualifier  \
90   2023-07-06              NaN        $18 USD           noche   
208  2023-08-22              NaN        $47 USD           noche   
193  2023-07-06              NaN        $47 USD           noche   
240  2023-07-06          $38 USD        $48 USD           noche   
216  2023-07-10          $48 USD        $55 USD           noche   
37   2023-07-06              NaN        $45 USD           noche   
146  2023-07-30              NaN        $40 USD           noche   
170  2023-07-06              NaN        $57 USD           noche   
168  2023-07-06              NaN       $350 USD           noche   
255  2023-07-07              NaN        $64 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna provincia  
90              1000      2023-06-24       2023-05-01  239.919355    CHUBUT  
208             1000      2023-06-24       2023-05-01  239.919355    CHUBUT  
193             1000      2023-06-24       2023-05-01  239.919355    CHUBUT  
240             1000      2023-06-24       2023-05-01  239.919355    CHUBUT  
216             1000      2023-06-24       2023-05-01  239.919355    CHUBUT  
37              1000      2023-06-24       2023-05-01  239.919355    CHUBUT  
146             1000      2023-06-24       2023-05-01  239.919355    CHUBUT  
170             1000      2023-06-24       2023-05-01  239.919355    CHUBUT  
168             1000      2023-06-24       2023-05-01  239.919355    CHUBUT  
255             1000      2023-06-24       202

In [3]:
# Carga de archivos de datos de Neuquen
df_neuquen = pd.read_csv('https://raw.githubusercontent.com/carlos1680/CPEDWObligatorio/main/Neuquen.csv')

df_neuquen["provincia"] = "NEUQUEN"

df_neuquen.sample(10)

id_alojamiento                                               name  \
216            37603595     5 - Pura tranquilidad en el corazón de Neuquén   
207            35734921     4 - Pura tranquilidad en el corazón de Neuquén   
24   803657303081651211                        Un oasis en Neuquén Capital   
144  691436380089538272  Cipolletti Casa250m2 C/Pileta, 8 Pax-Min 01 Se...   
97   600000166244200012                               Departamentos Leloir   
105  873742755590556492                        Hermoso monoambiente NUEVO!   
239            53511419                                Departamento Centro   
223            54181126  Hermoso estudio centro de Cipo! Amueblados Com...   
88             42634145                              Murmullos de Sueños.-   
10             54231128  Departamento amplio luminoso y hermoso en Neuquén   

         category     rating                    city   latitude  longitude  \
216   shared_room        NaN  Confluencia Department -38.951840 -68.058100   
207  private_room   3,75 (4)                 Neuquén -38.953170 -68.105900   
24    entire_home    5,0 (3)                 Neuquén -38.940095 -68.074063   
144   entire_home        NaN              Cipolletti -38.951141 -67.985878   
97    entire_home   4,33 (6)                 Neuquén -38.942795 -68.046227   
105   entire_home      Nuevo                 Neuquén -38.953140 -68.067270   
239   entire_home        NaN                 Neuquén -38.953380 -68.066910   
223   entire_home        NaN              Cipolletti -38.939580 -68.000750   
88    entire_home        NaN              Cipolletti -38.927950 -68.012980   
10    entire_home  4,86 (56)                 Neuquén -38.954930 -68.063000   

                                       characteristics    check_in  \
216  ['2 camas', '2 dormitorios', '2 baños comparti...  2023-07-01   
207  ['3 camas', '2 dormitorios', '1 cama doble', '...  2023-07-01   
24   ['5 camas', '3 dormitorios', '2,5 baños', '1 c...  2023-07-01   
144  ['6 camas', '3 dormitorios', '3 baños', 'Estac...  2023-07-01   
97   ['3 camas', '2 dormitorios', '1 ,5 baños', '1 ...  2023-07-03   
105  ['No hay evaluaciones', '1 cama', 'Monoambient...  2023-07-01   
239  ['No hay evaluaciones', '2 camas', '1 dormitor...  2023-07-01   
223  ['3 camas', 'Monoambiente', '1 baño', '2 camas...  2023-07-01   
88   ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-01   
10   ['3 camas', '2 dormitorios', '1 ,5 baños', '1 ...  2023-07-01   

      check_out price_discounted price_original price_qualifier  \
216  2023-07-06              NaN        $60 USD           noche   
207  2023-07-06              NaN        $55 USD           noche   
24   2023-07-06          $70 USD        $98 USD           noche   
144  2023-07-06              NaN       $180 USD           noche   
97   2023-07-08              NaN        $68 USD           noche   
105  2023-07-06          $38 USD        $48 USD           noche   
239  2023-07-06              NaN        $67 USD           noche   
223  2023-07-06              NaN        $50 USD           noche   
88   2023-07-06              NaN        $20 USD           noche   
10   2023-07-06              NaN        $45 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna provincia  
216              298      2023-06-24       2023-05-01  239.919355   NEUQUEN  
207              298      2023-06-24       2023-05-01  239.919355   NEUQUEN  
24               298      2023-06-24       2023-05-01  239.919355   NEUQUEN  
144              298      2023-06-24       2023-05-01  239.919355   NEUQUEN  
97               298      2023-06-24       2023-05-01  239.919355   NEUQUEN  
105              298      2023-06-24       2023-05-01  239.919355   NEUQUEN  
239              298      2023-06-24       2023-05-01  239.919355   NEUQUEN  
223              298      2023-06-24       2023-05-01  239.919355   NEUQUEN  
88               298      2023-06-24       2023-05-01  239.919355   NEUQUEN  
10

In [4]:
# Carga de archivos de datos de Rio Negro
df_rio_negro = pd.read_csv('https://raw.githubusercontent.com/carlos1680/CPEDWObligatorio/main/Rio_Negro.csv')

df_rio_negro["provincia"] = "RIO NEGRO"

df_rio_negro.sample(10)

id_alojamiento                                               name  \
9    897789125691156350   Hermosa Vista y Tranquilidad. Cercano al Centro.   
145            14053989                           CABAÑA CON COSTA DE LAGO   
147             5837517        Departamento LOS RODODENDROS  2 dormitorios   
56   801844068417930056                                    Cabaña Bittehus   
131            23005609         El Refugio Miralejos, cabaña en la montaña   
132            17009897   Aljabas, cabaña con vista al lago y a la montaña   
48   793355312470293540                           Cabaña en los kilómetros   
269            27133670                        Departamento sobre el Lago!   
212            32958687  JARDÍN LAKEANDVIEW CERCA DE la PLAYA LLAO LLAO...   
90             25286446                                    Casita Hillside   

        category      rating                     city  latitude  longitude  \
9    entire_home       Nuevo  San Carlos de Bariloche -41.13387  -71.28522   
145  entire_home  4,95 (133)  San Carlos de Bariloche -41.06827  -71.47169   
147  entire_home  4,51 (181)               Dina Huapi -41.07712  -71.17582   
56   entire_home     5,0 (6)  San Carlos de Bariloche -41.13118  -71.34478   
131  entire_home    4,9 (62)  San Martin de los Andes -40.16348  -71.28394   
132  entire_home   4,9 (134)                Los Lagos -40.72124  -71.68800   
48   entire_home   4,73 (11)  San Carlos de Bariloche -41.13316  -71.36912   
269  entire_home    4,9 (42)  San Carlos de Bariloche -41.13075  -71.28966   
212  entire_home     5,0 (7)  San Carlos de Bariloche -41.05039  -71.49948   
90   entire_home   4,94 (79)  San Carlos de Bariloche -41.04601  -71.51584   

                                       characteristics    check_in  \
9    ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-09-17   
145  ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-01   
147  ['1 cama', '2 dormitorios', '1 baño', '1 cama ...  2023-08-31   
56   ['1 cama', 'Monoambiente', '1 baño', 'Califica...  2023-07-02   
131  ['4 camas', '1 dormitorio', '1 cama doble', '3...  2023-07-01   
132  ['4 camas', '2 dormitorios', '2 baños', '1 cam...  2023-07-23   
48   ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-07-01   
269  ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-07-01   
212  ['1 cama', '1 dormitorio', '1 baño', 'Estacion...  2023-07-30   
90   ['1 cama', '1 dormitorio', '1 baño', '1 cama q...  2023-07-03   

      check_out price_discounted price_original price_qualifier  \
9    2023-09-22              NaN        $50 USD           noche   
145  2023-07-06          $95 USD       $120 USD           noche   
147  2023-09-05              NaN        $22 USD           noche   
56   2023-07-07              NaN        $70 USD           noche   
131  2023-07-06              NaN        $60 USD           noche   
132  2023-07-29         $139 USD       $170 USD           noche   
48   2023-07-06              NaN        $64 USD           noche   
269  2023-07-06              NaN        $93 USD           noche   
212  2023-08-04              NaN       $180 USD           noche   
90   2023-07-08              NaN       $120 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  provincia  
9               1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
145             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
147             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
56              1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
131             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
132             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
48              1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
269             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
212             1000      2023-06-24       2023-05-01  239.919355  RIO 

In [5]:
# Carga de archivos de datos de Tierra delFuego
df_tierra_del_fuego = pd.read_csv('https://raw.githubusercontent.com/carlos1680/CPEDWObligatorio/main/Tierra_del_Fuego.csv')

df_tierra_del_fuego["provincia"] = "TIERRA DEL FUEGO"

df_tierra_del_fuego.sample(10)

id_alojamiento                                           name  \
17   604911298357239989                    IBF SKY - Ushuaia Apartment   
143             2942774    Room with incredible sea and mountains view   
198  751515304138637905      Agradable cabaña con amplias dimensiones.   
48   916571989269028033                 Beautiful Apartment in Ushuaia   
222            53445794  Departamento con excelente ubicación y vistas   
138  889733753812060778                 Ushuaia Mágica - Ush AustraLis   
218             3395040                Amazing view, 4min away intown.   
241            15506599            DEPARTAMENTOS SAN MARTÍN - UNIDAD F   
188  751515304138637905      Agradable cabaña con amplias dimensiones.   
240  828090107927300389                                      GRUshuaia   

         category      rating     city   latitude  longitude  \
17    entire_home   4,93 (27)  Ushuaia -54.807125 -68.318497   
143  private_room  4,66 (228)  Ushuaia -54.795490 -68.313810   
198   entire_home         NaN  Tolhuin -54.507803 -67.201997   
48    entire_home       Nuevo  Ushuaia -54.810260 -68.318020   
222   entire_home   4,91 (22)  Ushuaia -54.808500 -68.320390   
138   entire_home       Nuevo  Ushuaia -54.811890 -68.326950   
218   entire_home       Nuevo  Ushuaia -54.812930 -68.326460   
241   entire_home    4,8 (41)  Ushuaia -54.805970 -68.306390   
188   entire_home         NaN  Tolhuin -54.507803 -67.201997   
240   entire_home         NaN  Ushuaia -54.815088 -68.329166   

                                       characteristics    check_in  \
17   ['1 cama', '1 dormitorio', '1 baño', '1 cama q...  2023-07-01   
143  ['2 camas', '1 dormitorio', '1 cama doble', '1...  2023-07-01   
198  ['8 camas', '3 dormitorios', '1 baño', 'Wifi',...  2023-07-01   
48   ['1 cama', '1 dormitorio', '1 baño', 'Wifi', '...  2023-07-01   
222  ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-06   
138  ['2 camas', 'Monoambiente', '1 baño', 'Wifi', ...  2023-07-09   
218  ['4 camas', '2 dormitorios', '1 baño', 'Wifi',...  2023-07-01   
241  ['4 camas', '1 dormitorio', '1 baño', 'Wifi', ...  2023-07-01   
188  ['8 camas', '3 dormitorios', '1 baño', 'Wifi',...  2023-07-01   
240  ['No hay evaluaciones', '1 cama', 'Monoambient...  2023-09-13   

      check_out price_discounted price_original price_qualifier  \
17   2023-07-06              NaN        $95 USD           noche   
143  2023-07-06              NaN        $28 USD           noche   
198  2023-07-06              NaN        $79 USD           noche   
48   2023-07-06              NaN        $30 USD           noche   
222  2023-07-12              NaN        $75 USD           noche   
138  2023-07-14              NaN        $45 USD           noche   
218  2023-07-06              NaN        $56 USD           noche   
241  2023-07-08              NaN        $85 USD           noche   
188  2023-07-06              NaN        $79 USD           noche   
240  2023-09-18          $51 USD        $85 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  \
17              1000      2023-06-24       2023-05-01  239.919355   
143             1000      2023-06-24       2023-05-01  239.919355   
198             1000      2023-06-24       2023-05-01  239.919355   
48              1000      2023-06-24       2023-05-01  239.919355   
222             1000      2023-06-24       2023-05-01  239.919355   
138             1000      2023-06-24       2023-05-01  239.919355   
218             1000      2023-06-24       2023-05-01  239.919355   
241             1000      2023-06-24       2023-05-01  239.919355   
188             1000      2023-06-24       2023-05-01  239.919355   
240             1000      2023-06-24       2023-05-01  239.919355   

            provincia  
17   TIERRA DEL FUEGO  
143  TIERRA DEL FUEGO  
198  TIERRA DEL FUEGO  
48   TIERRA DEL FUEGO  
222  TIERRA DEL FUEGO  
138  TIERRA DEL FUEGO  
218  TIERRA DEL FUEGO  
241  TIERRA DEL FUEGO  
188 

In [6]:
# Verificacion de columnas para concatenar
(df_chubut.columns == df_neuquen.columns) & (df_rio_negro.columns == df_tierra_del_fuego.columns) & (df_chubut.columns == df_tierra_del_fuego.columns)

array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True])

In [7]:
# Concatenacion de los dataframes separados por provincias en un unico archivo
df_alquileres = pd.concat([df_chubut, df_neuquen, df_rio_negro, df_tierra_del_fuego])
df_alquileres.sample(10)

id_alojamiento                                         name  \
183            48043650                          Latitud Sur Ushuaia   
161            24852036                           Mi Casa Es Su Casa   
219            49179716         Moderno departamento para 4 personas   
23   742686726143047393              Studio cerca del centro Neuquen   
143  599328787884286084                               Blues céntrico   
139  640943103810696995                         Cabaña Las Piedritas   
46             24851987              Carabelli Departamento Centrico   
147             5837517  Departamento LOS RODODENDROS  2 dormitorios   
115  597968065402649477  Casa Residencia artística Sueño del bosque.   
25   880037609372703201                                    Península   

        category      rating                     city   latitude  longitude  \
183  entire_home   4,95 (22)                  Ushuaia -54.810780 -68.327640   
161  entire_home    4,2 (20)               Rio Grande -53.783470 -67.702090   
219  entire_home    4,67 (3)               Cipolletti -38.948870 -67.983440   
23   entire_home    4,75 (4)                  Neuquén -38.943883 -68.046642   
143  entire_home    5,0 (53)  San Carlos de Bariloche -41.133860 -71.308370   
139  entire_home    5,0 (12)  San Martin de los Andes -40.124440 -71.321350   
46   entire_home     4,5 (6)               Rio Grande -53.785830 -67.700520   
147  entire_home  4,51 (181)               Dina Huapi -41.077120 -71.175820   
115  entire_home   4,85 (20)                  Ushuaia -54.803299 -68.307808   
25   entire_home       Nuevo                Bariloche -41.067385 -71.450768   

                                       characteristics    check_in  \
183  ['2 camas', 'Monoambiente', '1 baño', '1 cama ...  2023-07-01   
161  ['4 camas', '2 dormitorios', '1 baño', '4 cama...  2023-07-01   
219  ['4 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-08-17   
23   ['2 camas', 'Monoambiente', '1 baño', 'Estacio...  2023-07-01   
143  ['1 baño', '1 cama queen', '2 camas de una pla...  2023-09-02   
139  ['5 camas', '2 dormitorios', '1 ,5 baños', '1 ...  2023-07-01   
46   ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-07-01   
147  ['1 cama', '2 dormitorios', '1 baño', '1 cama ...  2023-08-31   
115  ['2 camas', '1 dormitorio', '1 baño', '2 camas...  2023-07-08   
25        ['1 cama', '1 dormitorio', '1 baño', 'Wifi']  2023-07-30   

      check_out price_discounted price_original price_qualifier  \
183  2023-07-06              NaN        $59 USD           noche   
161  2023-07-06              NaN        $28 USD           noche   
219  2023-08-22              NaN        $47 USD           noche   
23   2023-07-06              NaN        $24 USD           noche   
143  2023-09-08         $180 USD       $200 USD           noche   
139  2023-07-06              NaN       $140 USD           noche   
46   2023-07-07              NaN        $44 USD           noche   
147  2023-09-05              NaN        $22 USD           noche   
115  2023-07-13              NaN        $39 USD           noche   
25   2023-08-04              NaN       $180 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  \
183             1000      2023-06-24       2023-05-01  239.919355   
161             1000      2023-06-24       2023-05-01  239.919355   
219              298      2023-06-24       2023-05-01  239.919355   
23               298      2023-06-24       2023-05-01  239.919355   
143             1000      2023-06-24       2023-05-01  239.919355   
139             1000      2023-06-24       2023-05-01  239.919355   
46              1000      2023-06-24       2023-05-01  239.919355   
147             1000      2023-06-24       2023-05-01  239.919355   
115             1000      2023-06-24       2023-05-01  239.919355   
25              1000      2023-06-24       2023-05-01  239.919355   

            provincia  
183  TIERRA DEL FUEGO  
161  TIERRA DEL FUEGO  
219  

In [8]:
# Reseteo del index, permitiendo duplicados porque aun no sabemos si son unicos los valores
# df_alquileres.reset_index(names = 'id_alojamiento', drop=True,inplace=True,allow_duplicates= True)
# df_alquileres.reset_index(names = 'id_alojamiento', drop=True,inplace=True,allow_duplicates= False)
df_alquileres.reset_index(drop=True,inplace=True)

*Verificacion de filas de los dataframes*

In [9]:
# Verificacion de la cantidad de filas de cada dataframe
print("La cantidad de filas del dataframe de Chubut es:"+str(df_chubut.shape[0]))
print("La cantidad de filas del dataframe de Neuquen es:"+str(df_neuquen.shape[0]))
print("La cantidad de filas del dataframe de Rio Negro es:"+str(df_rio_negro.shape[0]))
print("La cantidad de filas del dataframe de Tierra del fuego es:" + str(df_tierra_del_fuego.shape[0]))
print("La cantidad de filas del dataframe unificado de alquileres es:" + str(df_alquileres.shape[0]))


La cantidad de filas del dataframe de Chubut es:270
La cantidad de filas del dataframe de Neuquen es:270
La cantidad de filas del dataframe de Rio Negro es:270
La cantidad de filas del dataframe de Tierra del fuego es:270
La cantidad de filas del dataframe unificado de alquileres es:1080


*Verificacion de las columnas de los dataframes*

In [10]:
# Verificacion de la cantidad de columnas de cada dataframe
print("La cantidad de columnas del dataframe de Chubut es:" +
      str(df_chubut.shape[1]))
print("La cantidad de columnas del dataframe de Neuquen es:" +
      str(df_neuquen.shape[1]))
print("La cantidad de columnas del dataframe de Rio Negro es:" +
      str(df_rio_negro.shape[1]))
print("La cantidad de columnas del dataframe de Tierra del fuego es:" +
      str(df_tierra_del_fuego.shape[1]))
print("La cantidad de columnas del dataframe unificado de alquileres es:" +
      str(df_alquileres.shape[1]))

La cantidad de columnas del dataframe de Chubut es:18
La cantidad de columnas del dataframe de Neuquen es:18
La cantidad de columnas del dataframe de Rio Negro es:18
La cantidad de columnas del dataframe de Tierra del fuego es:18
La cantidad de columnas del dataframe unificado de alquileres es:18


*Verificacion general e inicial del dataframe unificado*

In [11]:
# Informacion de las columnas, sus tipos y faltantes
df_alquileres.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1080 entries, 0 to 1079
Data columns (total 18 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   id_alojamiento    1080 non-null   int64  
 1   name              1080 non-null   object 
 2   category          1080 non-null   object 
 3   rating            886 non-null    object 
 4   city              1076 non-null   object 
 5   latitude          1080 non-null   float64
 6   longitude         1080 non-null   float64
 7   characteristics   1080 non-null   object 
 8   check_in          1080 non-null   object 
 9   check_out         1080 non-null   object 
 10  price_discounted  201 non-null    object 
 11  price_original    1080 non-null   object 
 12  price_qualifier   1080 non-null   object 
 13  cant_resultados   1080 non-null   int64  
 14  fecha_scrapping   1080 non-null   object 
 15  fecha_cambio_bna  1080 non-null   object 
 16  cambio_bna        1080 non-null   float64


In [12]:
# Informacion de las columnas, sus tipos y faltantes
df_alquileres.describe()

id_alojamiento     latitude    longitude  cant_resultados   cambio_bna
count    1.080000e+03  1080.000000  1080.000000      1080.000000  1080.000000
mean     3.685341e+17   -44.426201   -68.723924       824.500000   239.919355
std      3.903051e+17     6.164758     2.159715       304.115744     0.000000
min      4.468310e+05   -54.844914   -71.688000       298.000000   239.919355
25%      3.529220e+07   -47.854503   -71.310280       824.500000   239.919355
50%      5.408344e+07   -42.003780   -68.305610      1000.000000   239.919355
75%      7.872495e+17   -39.609811   -68.049328      1000.000000   239.919355
max      9.198428e+17   -38.874370   -62.920960      1000.000000   239.919355

In [13]:
# Informacion de las columnas, sus tipos y faltantes
df_alquileres.duplicated(keep="first").sum()

51

In [14]:
# Informacion de las columnas, sus tipos y faltantes
df_alquileres.isna().sum()

id_alojamiento        0
name                  0
category              0
rating              194
city                  4
latitude              0
longitude             0
characteristics       0
check_in              0
check_out             0
price_discounted    879
price_original        0
price_qualifier       0
cant_resultados       0
fecha_scrapping       0
fecha_cambio_bna      0
cambio_bna            0
provincia             0
dtype: int64

*Conclusiones iniciales*
<br>
+ El dataframe unificado, coincide en cantidad de filas y columnas correspondientes a los dataframes individuales
+ Debemos de corregir algunos tipos de datos de columnas
+ Tenemos algunas columnas para reescalar
+ Tenemos filas duplicadas para revisar
+ Tenemos valores nulos para revisar, en principio para las columnas rating y price_discounted

**<h1>2. Análisis de duplicados</h1>**

*Analisis y tratamiento de duplicados global*

In [15]:
# Verifico los duplicados, que coinciden en toda la fila
df_alquileres.duplicated(keep="first").sum()

51

In [16]:
#  Listo los valores duplicados
df_alquileres[df_alquileres.duplicated(keep="first")]

id_alojamiento                                               name  \
54              40755147  La Amarilla Casa de Huéspedes (con cocina) Rut...   
72    586216477715336949  Encantador dpto. con vista al mar en Puerto Ma...   
90              37836757                                     Rincón del Sol   
113   736253141994173324           Refugio Ciprés con tina de agua caliente   
127             21925703      Hermoso Duplex con Patio y Parrilla. (4 pax).   
128   644043911246503372     Los Molles, casa familiar  en entorno natural.   
129   883174783781736766                             Espacio 2 - Rada Tilly   
138             53298317                       Mar de Olivillos Planta alta   
162             31081731                 Domos Puelo, para dos, con jacuzzi   
182             50704005   Hostel patagónico con vista 360 de la Cordillera   
189   800916516894617209  Cómodas Habitaciones, 50 m del Mar con Desayun...   
216             53336620               Comodo Monoambiente en puerto madryn   
221             54362650     "Ty Bychan" \nLa Casita. Entre raíces galesas.   
234   807474929736554545                          Cabaña de Playa. Bajada 7   
247             41936762                                       El Tornado B   
253   909336244153347770                                      Bella Vista 2   
255   803666373797660065                      Casa nueva equipada con patio   
265   914317237704080953                    Dep,AlquilerTemporario,Acogedor   
630             25286446                                    Casita Hillside   
720              4635954  ESTUDIO A 20 METROS DEL CIVICO CENTER, VISTA A...   
756             38777312                                  Cabañas Waldesruh   
757             37245663  Preciosa cabaña en el bosque de Llaollao con v...   
828   604911298357239989                        IBF SKY - Ushuaia Apartment   
832   711705563371014313                 C) Cama en cuarto mixto compartido   
846             39446425                  Cozy Flat, with incredible views.   
850             54314756                       Infinity View Pent House III   
852   604911298357239989                        IBF SKY - Ushuaia Apartment   
882   710167445018555343        Apartamento Amundsen 6J - Estudio - Ushuaia   
885             40366875                 H) Cama en cuarto mixto compartido   
893   755135392594127489  Encantadoras vistas del canal de beagle - Amun...   
922             38919535       Moderno y luminoso estudio en  pleno centro.   
943   570310791164532435                              Casa barrio de sueños   
954              2942774        Room with incredible sea and mountains view   
971             24852036                                 Mi Casa Es Su Casa   
989             53797805            Humilde hogar, en el corazón de Ushuaia   
993             48043650                                Latitud Sur Ushuaia   
1001  776661041221694371                   Lindo estudio céntrico y moderno   
1008  751515304138637905          Agradable cabaña con amplias dimensiones.   
1009  837061192294336671                             Loft del Fin del Mundo   
1010  847640105410806361                                          Mini casa   
1011            54301035  Departamento de 2 ambientes -vista al Canal Be...   
1017  831611894265914063                                       Ushuaia home   
1018            54258723                NEW APARTMENT - LAS CUMBRES USHUAIA   
1032            53445794      Departamento con excelente ubicación y vistas   
1046  626675407573680894                            Precioso Departamento A   
1054  906355891628807276                                        Grand House   
1060  858196681965117698                           La Tempranera del Beagle   
1062  562143078127144674                                           Turquesa   
1063            51759643     Departamento con vista abierta  a las montañas   
1064             8048378               DESAPARTAMENTO SAN MA

In [17]:
# Elimino los duplicados, que coinciden en toda la fila
df_alquileres.drop_duplicates(keep="first", inplace=True)
df_alquileres.shape

(1029, 18)

*Analisis y tratamiento de duplicados por columna*

+ Columna 1: id_alojamiento

In [18]:
# Busco si hay duplicados
df_alquileres['id_alojamiento'].duplicated(keep='first').sum()

1

In [19]:
# Identifico filas con esos valores duplicados
df_alquileres[df_alquileres['id_alojamiento'].duplicated(keep='first')]

id_alojamiento          name     category    rating  \
648        53729814  Pequeña casa  entire_home  5,0 (36)   

                        city  latitude  longitude  \
648  San Carlos de Bariloche -41.14768  -71.31827   

                                       characteristics    check_in  \
648  ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-08-18   

      check_out price_discounted price_original price_qualifier  \
648  2023-08-23              NaN        $39 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  provincia  
648             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO

In [20]:
# Muestro las filas que corresponden a los valores duplicados, todas
df_alquileres[df_alquileres['id_alojamiento'] == 53729814]

id_alojamiento          name     category    rating  \
647        53729814  Pequeña casa  entire_home  5,0 (36)   
648        53729814  Pequeña casa  entire_home  5,0 (36)   

                        city  latitude  longitude  \
647  San Carlos de Bariloche -41.14768  -71.31827   
648  San Carlos de Bariloche -41.14768  -71.31827   

                                       characteristics    check_in  \
647  ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-08-18   
648  ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-08-18   

      check_out price_discounted price_original price_qualifier  \
647  2023-08-23              NaN        $39 USD           noche   
648  2023-08-23              NaN        $39 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  provincia  
647             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
648             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO

-> Analisis de los valores de las filas duplicadas.

In [21]:
#Verificamos los valores 
df_alquileres.loc[647]

id_alojamiento                                               53729814
name                                                     Pequeña casa
category                                                  entire_home
rating                                                       5,0 (36)
city                                          San Carlos de Bariloche
latitude                                                    -41.14768
longitude                                                   -71.31827
characteristics     ['1 cama', '1 dormitorio', '1 baño', '1 cama d...
check_in                                                   2023-08-18
check_out                                                  2023-08-23
price_discounted                                                  NaN
price_original                                                $39 USD
price_qualifier                                                 noche
cant_resultados                                                  1000
fecha_scrapping     

In [22]:
# Verificamos los valores
df_alquileres.loc[648]

id_alojamiento                                               53729814
name                                                     Pequeña casa
category                                                  entire_home
rating                                                       5,0 (36)
city                                          San Carlos de Bariloche
latitude                                                    -41.14768
longitude                                                   -71.31827
characteristics     ['1 cama', '1 dormitorio', '1 baño', '1 cama d...
check_in                                                   2023-08-18
check_out                                                  2023-08-23
price_discounted                                                  NaN
price_original                                                $39 USD
price_qualifier                                                 noche
cant_resultados                                                  1000
fecha_scrapping     

In [23]:
# Verificamos los valores para la variable que tendria fiferencia
df_alquileres.loc[647]["characteristics"]

"['1 cama', '1 dormitorio', '1 baño', '1 cama doble', 'Estacionamiento gratis', 'Wifi']"

In [24]:
# Verificamos los valores para la variable que tendria fiferencia
df_alquileres.loc[648]["characteristics"]

"['1 cama', '1 dormitorio', '1 baño', '1 cama doble', 'Estacionamiento gratis', 'Wifi', 'Calificado con 5 de 5 en limpieza']"

*ACCION:*
<BR>
*Vemos que seria el mismo registro con excepcion de la serie characteristics, donde tiene agregada comentario de calificacion*
<br>
*Como consideramos que cada valor de id_alojamiento, deberia ser unico, eliminamos el que consideramos repetido y menos se adecua a los valores esperados por serie*

In [25]:
# Descartamos el registro que no tiene agregado el comentario de la calificacion
df_alquileres.drop([648], inplace=True)
df_alquileres[df_alquileres['id_alojamiento'] == 53729814]

id_alojamiento          name     category    rating  \
647        53729814  Pequeña casa  entire_home  5,0 (36)   

                        city  latitude  longitude  \
647  San Carlos de Bariloche -41.14768  -71.31827   

                                       characteristics    check_in  \
647  ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-08-18   

      check_out price_discounted price_original price_qualifier  \
647  2023-08-23              NaN        $39 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  provincia  
647             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO

+ Columna 2: name

In [26]:
# Verifico los duplicados, que coinciden en toda la fila
df_alquileres['name'].duplicated(keep="first").sum()

1

In [27]:
# Verifico los duplicados, que coinciden en toda la fila
df_alquileres[df_alquileres['name'].duplicated(keep="first")]

id_alojamiento                                         name  \
794  800041160997364459  Vista al Lago en el Bosque - Hab Compartida   

        category   rating                     city  latitude  longitude  \
794  shared_room  5,0 (8)  San Carlos de Bariloche  -41.1333  -71.36914   

                                       characteristics    check_in  \
794  ['1 cama', '1 dormitorio', '1 baño compartido'...  2023-07-05   

      check_out price_discounted price_original price_qualifier  \
794  2023-07-10              NaN        $23 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  provincia  
794             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO

In [28]:
df_alquileres[df_alquileres['name'].str.contains('Vista al Lago en el Bosque - Hab Compartida')]

id_alojamiento                                         name  \
712  797077259992479162  Vista al Lago en el Bosque - Hab Compartida   
794  800041160997364459  Vista al Lago en el Bosque - Hab Compartida   

        category   rating                     city  latitude  longitude  \
712  shared_room  5,0 (8)  San Carlos de Bariloche -41.13455  -71.36807   
794  shared_room  5,0 (8)  San Carlos de Bariloche -41.13330  -71.36914   

                                       characteristics    check_in  \
712  ['1 cama', '1 dormitorio', '1 baño compartido'...  2023-07-11   
794  ['1 cama', '1 dormitorio', '1 baño compartido'...  2023-07-05   

      check_out price_discounted price_original price_qualifier  \
712  2023-07-18              NaN        $21 USD           noche   
794  2023-07-10              NaN        $23 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  provincia  
712             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
794             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO

*ACCION:*
<br>
*En este caso vemos que los nombre coinciden pero, pueden corresponder a diferentes lugares, de hecho, estan sobre diferentes calles, segun sus coordenadas.*
<br>
*Por tal motivo los dejamos a ambos registros*

+ Columna 3: category

In [29]:
# Verifico los duplicados, que coinciden en toda la fila
df_alquileres['category'].duplicated(keep="first").sum()

1025

In [30]:
# Listo los duplicados segun category
df_alquileres[df_alquileres['category'].duplicated(keep="first")].sample(10)

id_alojamiento                                               name  \
309            33722564     2 - Pura tranquilidad en el corazón de Neuquén   
437            48145206        11-E Espectacular departamento Alto Neuquen   
647            53729814                                       Pequeña casa   
520            49091963                        Rústico, prolijo, acogedor.   
64   833175601621381109                    Mono ambiente a metros del mar.   
761             5684334  Gardeners Delight Studio La Casita Barrio Chap...   
982  690413283545175725                                      Las Aguilas 3   
471            53105919            CENTRO DE LOFT!! ALTO COMAHUE COMPRAS!!   
560  876704382378753640      Encantadora cabaña rústica con acceso al lago   
339  688433801108777336                         zona residencial, y segura   

        category      rating                     city   latitude  longitude  \
309  shared_room   4,72 (25)                  Neuquén -38.953620 -68.103490   
437  entire_home         NaN                  Neuquén -38.943990 -68.047370   
647  entire_home    5,0 (36)  San Carlos de Bariloche -41.147680 -71.318270   
520  entire_home         NaN                  Neuquén -38.945140 -68.081720   
64   entire_home     5,0 (7)            Puerto Madryn -42.772080 -65.030160   
761  entire_home  4,94 (109)  San Martín de los Andes -40.137070 -71.281610   
982  entire_home    4,67 (9)                  Ushuaia -54.814922 -68.341647   
471  entire_home         NaN                  Neuquén -38.943040 -68.062740   
560  entire_home       Nuevo  San Carlos de Bariloche -41.167240 -71.416200   
339  entire_home    5,0 (10)                  Neuquén -38.949511 -68.104122   

                                       characteristics    check_in  \
309  ['1 cama', '1 dormitorio', '2 baños compartido...  2023-07-01   
437  ['No hay evaluaciones', '2 camas', '1 dormitor...  2023-07-01   
647  ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-08-18   
520  ['No hay evaluaciones', '1 cama', '1 dormitori...  2023-08-12   
64   ['1 cama', '1 dormitorio', '1 baño', 'Califica...  2023-07-01   
761  ['2 camas', '1 dormitorio', '1 baño', '2 camas...  2023-07-01   
982  ['4 camas', '1 dormitorio', '1 baño', 'Wifi', ...  2023-07-08   
471  ['No hay evaluaciones', '2 camas', 'Monoambien...  2023-07-01   
560  ['1 cama', '1 dormitorio', '1 baño', '1 cama q...  2023-07-01   
339  ['3 camas', '1 dormitorio', '1 baño', 'Estacio...  2023-07-23   

      check_out price_discounted price_original price_qualifier  \
309  2023-07-06              NaN        $12 USD           noche   
437  2023-07-06              NaN        $59 USD           noche   
647  2023-08-23              NaN        $39 USD           noche   
520  2023-08-18              NaN        $23 USD           noche   
64   2023-07-06              NaN        $54 USD           noche   
761  2023-07-06              NaN        $49 USD           noche   
982  2023-07-14              NaN        $50 USD           noche   
471  2023-07-06              NaN        $20 USD           noche   
560  2023-07-06              NaN        $69 USD           noche   
339  2023-07-28              NaN        $34 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  \
309              298      2023-06-24       2023-05-01  239.919355   
437              298      2023-06-24       2023-05-01  239.919355   
647             1000      2023-06-24       2023-05-01  239.919355   
520              298      2023-06-24       2023-05-01  239.919355   
64              1000      2023-06-24       2023-05-01  239.919355   
761             1000      2023-06-24       2023-05-01  239.919355   
982             1000      2023-06-24       2023-05-01  239.919355   
471              298      2023-06-24       2023-05-01  239.919355   
560             1000      2023-06-24       2023-05-01  239.919355   
339              298      2023-06-24       2023-05-01  239.919355   

           

*ACCION:*
<br>
*Entendemos que los valores de categoria se pueden repetir para los diferentes alquileres*
<br>
*por lo cual, no realizamos ninguna accion en base a esta columna*

+ Columna 4: rating

In [31]:
# Verifico los duplicado
df_alquileres['rating'].duplicated(keep="first").sum()

564

In [32]:
# Listamos algunas filas, repetidas por rating
df_alquileres[df_alquileres['rating'].duplicated(keep="first")].sample(10)

id_alojamiento                                          name  \
122   782580256541651315               Casa familiar con cálido jardín   
1053  638425013799679161  Departamento en Ushuaia totalmente equipado.   
125   883174783781736766                        Espacio 2 - Rada Tilly   
716   782060930905714812              Cabaña en la montaña: La Cascada   
448             30049426                          DEPARTAMENTO LIMAY I   
498   864683273673028486             Dpto, Micro centro, N/ managerial   
404   544914706745066154   Plaza Caviahue Apartment en Neuquén Capital   
328   699352390023248021                                        Duplex   
1076            26215568                          Hospedaje UshuaiAndo   
942   692111476259511524                                Vivir y soñar!   

          category     rating                     city   latitude  longitude  \
122   private_room    5,0 (4)                   Rawson -43.295951 -65.085494   
1053   entire_home   4,67 (9)                  Ushuaia -54.808900 -68.319470   
125    entire_home      Nuevo               Rada Tilly -45.925810 -67.575650   
716    entire_home        NaN  San Martin de los Andes -40.144050 -71.275120   
448    entire_home        NaN                  Neuquén -38.962880 -68.028220   
498    entire_home        NaN                  Neuquén -38.949020 -68.061780   
404    entire_home        NaN                  Neuquén -38.942020 -68.054690   
328    entire_home      Nuevo                  Neuquén -38.944580 -68.074130   
1076   entire_home  4,82 (50)                  Ushuaia -54.800790 -68.304230   
942    entire_home  4,91 (22)                  Ushuaia -54.809630 -68.320920   

                                        characteristics    check_in  \
122   ['1 ,5 baños', '1 cama doble', '2 camas de una...  2023-07-01   
1053  ['3 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-01   
125                ['1 cama', '1 dormitorio', '1 baño']  2023-07-01   
716   ['1 cama', '1 dormitorio', '1 baño', 'Estacion...  2023-07-01   
448   ['3 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-02   
498   ['No hay evaluaciones', '1 cama', '1 dormitori...  2023-07-01   
404       ['2 camas', '1 dormitorio', '1 baño', 'Wifi']  2023-09-14   
328   ['2 camas', '2 dormitorios', '2 baños', 'Estac...  2023-07-01   
1076  ['3 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-23   
942   ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-10-01   

       check_out price_discounted price_original price_qualifier  \
122   2023-07-06              NaN        $36 USD           noche   
1053  2023-07-06              NaN        $83 USD           noche   
125   2023-07-06              NaN        $52 USD           noche   
716   2023-07-06          $45 USD        $75 USD           noche   
448   2023-07-07              NaN        $36 USD           noche   
498   2023-07-06              NaN        $77 USD           noche   
404   2023-09-20              NaN        $50 USD           noche   
328   2023-07-06          $36 USD        $45 USD           noche   
1076  2023-07-28          $45 USD        $50 USD           noche   
942   2023-10-06              NaN        $70 USD           noche   

      cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  \
122              1000      2023-06-24       2023-05-01  239.919355   
1053             1000      2023-06-24       2023-05-01  239.919355   
125              1000      2023-06-24       2023-05-01  239.919355   
716              1000      2023-06-24       2023-05-01  239.919355   
448               298      2023-06-24       2023-05-01  239.919355   
498               298      2023-06-24       2023-05-01  239.919355   
404               298      2023-06-24       2023-05-01  239.919355   
328               298      2023-06-24       2023-05-01  239.919355   
1076             1000      2023-06-24       2023-05-01  239.919355   
942              1000      2023-06-24       2023-05-01  239.919355   

            

In [33]:
df_alquileres['rating'].unique()

array([nan, '5,0 (6)', '4,91 (70)', 'Nuevo', '4,96 (95)', '4,86 (7)',
       '4,82 (17)', '4,96 (75)', '4,91 (67)', '4,57 (14)', '4,6 (5)',
       '4,94 (18)', '4,95 (22)', '4,92 (64)', '4,94 (109)', '4,77 (44)',
       '4,89 (35)', '5,0 (7)', '4,66 (53)', '5,0 (34)', '4,98 (93)',
       '4,96 (51)', '4,82 (66)', '4,8 (51)', '4,83 (36)', '4,9 (20)',
       '5,0 (17)', '4,65 (49)', '4,7 (33)', '4,89 (9)', '4,97 (90)',
       '5,0 (10)', '4,79 (30)', '4,86 (74)', '4,86 (44)', '4,63 (128)',
       '5,0 (3)', '4,88 (16)', '4,81 (32)', '5,0 (9)', '4,93 (27)',
       '4,92 (13)', '4,5 (4)', '4,77 (13)', '4,67 (12)', '4,9 (176)',
       '4,95 (19)', '4,8 (15)', '5,0 (14)', '4,79 (14)', '5,0 (4)',
       '5,0 (19)', '4,67 (6)', '4,82 (137)', '5,0 (11)', '4,77 (102)',
       '4,47 (55)', '4,76 (134)', '4,74 (19)', '4,96 (27)', '4,91 (11)',
       '4,93 (15)', '4,62 (13)', '4,55 (33)', '4,4 (5)', '4,72 (25)',
       '4,58 (76)', '4,73 (11)', '4,83 (60)', '4,93 (28)', '4,7 (10)',
       '4,76 (85

*ACCION:*
<br>
*Entendemos que los valores de rating se pueden repetir para los diferentes alquileres*
<br>
*Por lo cual, dejamos los registros retpetidos segun esta columna*

+ Columna 5: city

In [34]:
# Verifico los duplicados, que coinciden en toda la fila
df_alquileres['city'].duplicated(keep="first").sum()

947

In [35]:
# Listamos repetidos, por city
df_alquileres[df_alquileres['city'].duplicated(keep="first")]

id_alojamiento                                               name  \
1               51484364     Hermoso Monoambiente con Vista al Mar y Pileta   
2     800749301724167838                           Departamento Bella Vista   
3               50769917  New Lil’ & Cozy - 100mts de la playa Puerto Ma...   
4     916738743930751566                                        Front Beach   
5               27877802      Luxury top 11th floor with the best 360 views   
...                  ...                                                ...   
1075  771725397497773450                            Casa Fueguina en Centro   
1076            26215568                               Hospedaje UshuaiAndo   
1077            42486711       Yeska C Dpto. Supercéntrico. Casco Histórico   
1078            51921207                                             KREN I   
1079  842195865716596989                                   Bahía de Ushuaia   

         category     rating           city   latitude  longitude  \
1     entire_home        NaN  Puerto Madryn -42.755775 -65.037265   
2     entire_home    5,0 (6)  Puerto Madryn -42.768007 -65.033108   
3     entire_home  4,91 (70)  Puerto Madryn -42.777590 -65.025890   
4     entire_home      Nuevo  Puerto Madryn -42.770540 -65.030810   
5     entire_home  4,96 (95)  Puerto Madryn -42.768330 -65.032100   
...           ...        ...            ...        ...        ...   
1075  entire_home  4,53 (17)        Ushuaia -54.803310 -68.302580   
1076  entire_home  4,82 (50)        Ushuaia -54.800790 -68.304230   
1077  entire_home   4,88 (8)        Ushuaia -54.805430 -68.308710   
1078  entire_home  4,85 (26)        Ushuaia -54.806170 -68.317800   
1079  entire_home        NaN        Ushuaia -54.793450 -68.276470   

                                        characteristics    check_in  \
1     ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-21   
2     ['3 camas', '1 dormitorio', '1 baño', 'Calific...  2023-08-01   
3     ['1 cama', '1 dormitorio', '1 baño', '1 cama q...  2023-07-09   
4                 ['2 camas', 'Monoambiente', '1 baño']  2023-07-01   
5     ['4 camas', '2 dormitorios', '2,5 baños', '1 c...  2023-07-06   
...                                                 ...         ...   
1075  ['3 camas', '2 dormitorios', '1 baño', 'Wifi',...  2023-08-14   
1076  ['3 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-23   
1077  ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-07-01   
1078  ['2 camas', 'Monoambiente', '1 baño', 'Wifi', ...  2023-07-10   
1079  ['1 cama', '1 dormitorio', '1 baño', 'Wifi', '...  2023-08-05   

       check_out price_discounted price_original price_qualifier  \
1     2023-07-26          $18 USD        $24 USD           noche   
2     2023-08-08              NaN        $43 USD           noche   
3     2023-07-14              NaN        $41 USD           noche   
4     2023-07-06              NaN        $35 USD           noche   
5     2023-07-11              NaN       $122 USD           noche   
...          ...              ...            ...             ...   
1075  2023-08-19          $60 USD        $76 USD           noche   
1076  2023-07-28          $45 USD        $50 USD           noche   
1077  2023-07-06              NaN        $63 USD           noche   
1078  2023-07-15              NaN        $55 USD           noche   
1079  2023-08-10              NaN        $49 USD           noche   

      cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  \
1                1000      2023-06-24       2023-05-01  239.919355   
2                1000      2023-06-24       2023-05-01  239.919355   
3                1000      2023-06-24       2023-05-01  239.919355   
4                1000      2023-06-24       2023-05-01  239.919355   
5                1000      2023-06-24       2023-05-01  239.919355   
...               ...             ...              ...         ...   
1075             1000      2023-06-24       2023-05-01  239.919355   
10

*ACCION:*
<br>
*Entendemos que la columna city, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 6: latitude

In [36]:
# Verificamos los duplicados
df_alquileres['latitude'].duplicated(keep="first").sum()

36

In [37]:
# Listamos algunos de los duplicados
df_alquileres[df_alquileres['latitude'].duplicated(keep="first")].sample(10)

id_alojamiento                                               name  \
296   661388148387143050  Apart céntrico 2 dorm  c/ balcon  cochera opci...   
1031  711709136406452484                 F) Cama en cuarto mixto compartido   
709             17846750           Cabaña en Bariloche a minutos del Centro   
243   909336244153347770                                      Bella Vista 2   
941             53903647                                          Skyline 4   
717             49546396          Departamento para 2. Excelente ubicación.   
364   810083074082367782                     Departamento luminoso centrico   
495             36808035                                       Las Martínez   
427   651446721456756885        Depto hermoso con estacionamiento incluido.   
1019            53445794      Departamento con excelente ubicación y vistas   

         category      rating                     city   latitude  longitude  \
296   entire_home     5,0 (6)                  Neuquén -38.951410 -68.061740   
1031  shared_room    4,67 (9)                  Ushuaia -54.796494 -68.298038   
709   entire_home  4,89 (108)  San Carlos de Bariloche -41.127640 -71.358010   
243   entire_home       Nuevo                   Esquel -42.917491 -71.294685   
941   entire_home    4,88 (8)                  Ushuaia -54.807163 -68.318520   
717   entire_home   4,89 (53)  San Carlos de Bariloche -41.134550 -71.299200   
364   entire_home         NaN                  Neuquén -38.949560 -68.054970   
495   entire_home    4,89 (9)                  Neuquén -38.952770 -68.067640   
427   entire_home    4,83 (6)                  Neuquén -38.949400 -68.052530   
1019  entire_home   4,91 (22)                  Ushuaia -54.808500 -68.320390   

                                        characteristics    check_in  \
296   ['3 camas', '2 dormitorios', '1 ,5 baños', 'Wi...  2023-07-01   
1031  ['1 cama', '1 dormitorio', '1,5 baños comparti...  2023-07-01   
709   ['4 camas', '2 dormitorios', '1 baño', '1 cama...  2023-08-09   
243               ['4 camas', '1 dormitorio', '1 baño']  2023-07-01   
941   ['1 cama', '1 dormitorio', '1 baño', 'Wifi', '...  2023-07-04   
717   ['1 cama', 'Monoambiente', '1 baño', '1 cama q...  2023-07-07   
364   ['No hay evaluaciones', '1 cama', 'Monoambient...  2023-10-02   
495   ['3 camas', '2 dormitorios', '1 baño', '1 cama...  2023-10-01   
427   ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-01   
1019  ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-06   

       check_out price_discounted price_original price_qualifier  \
296   2023-07-08              NaN        $50 USD           noche   
1031  2023-07-06              NaN        $12 USD           noche   
709   2023-08-14          $75 USD        $85 USD           noche   
243   2023-07-06          $52 USD        $65 USD           noche   
941   2023-07-09          $88 USD        $98 USD           noche   
717   2023-07-13              NaN        $70 USD           noche   
364   2023-10-07              NaN        $22 USD           noche   
495   2023-10-06              NaN        $70 USD           noche   
427   2023-07-06              NaN        $59 USD           noche   
1019  2023-07-12              NaN        $75 USD           noche   

      cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  \
296               298      2023-06-24       2023-05-01  239.919355   
1031             1000      2023-06-24       2023-05-01  239.919355   
709              1000      2023-06-24       2023-05-01  239.919355   
243              1000      2023-06-24       2023-05-01  239.919355   
941              1000      2023-06-24       2023-05-01  239.919355   
717              1000      2023-06-24       2023-05-01  239.919355   
364               298      2023-06-24       2023-05-01  239.919355   
495               298      2023-06-24       2023-05-01  239.919355   
427               298      2023-06-24       2023-05-01  239.919355   
1019             1000      2

In [38]:
# Vemos un ejemplo
df_alquileres[df_alquileres['latitude'] == -54.803600]

id_alojamiento                                               name  \
934             31825003  Departamento con inmejorable ubicación en Ushuaia   
1025  623178032761843625            departamento céntrico, cálido y cómodo.   

         category     rating     city  latitude  longitude  \
934   entire_home  4,9 (108)  Ushuaia  -54.8036  -68.30374   
1025  entire_home  4,88 (16)      AIU  -54.8036  -68.32132   

                                        characteristics    check_in  \
934   ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-01   
1025  ['3 camas', '1 dormitorio', '1 baño', 'Wifi', ...  2023-07-09   

       check_out price_discounted price_original price_qualifier  \
934   2023-07-06              NaN        $85 USD           noche   
1025  2023-07-14              NaN        $33 USD           noche   

      cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  \
934              1000      2023-06-24       2023-05-01  239.919355   
1025             1000      2023-06-24       2023-05-01  239.919355   

             provincia  
934   TIERRA DEL FUEGO  
1025  TIERRA DEL FUEGO

*ACCION:*
<br>
*Entendemos que la columna latitude, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 7: longitude

In [39]:
# Verificamos los duplicados
df_alquileres['longitude'].duplicated(keep="first").sum()

20

In [40]:
# Listamos algunos de los duplicados
df_alquileres[df_alquileres['longitude'].duplicated(keep="first")].sample(10)

id_alojamiento                                               name  \
1044            51921353                                            KREN II   
86              51311028   Amplio Monoambiente con estacionamiento,cerrado.   
1071  700712866756917764                            D) Cuarto doble privado   
531   833878451226342434                      El lugar de Ana - Am. Comahue   
851   751455808581955668                                          Ushuaia54   
527             32223913                     Mono-Leloir AMUEBLADOS COMAHUE   
257             53295849  Preciosa unidad de alquiler de 3 dormitorios e...   
948   889733753812060778                     Ushuaia Mágica - Ush AustraLis   
243   909336244153347770                                      Bella Vista 2   
941             53903647                                          Skyline 4   

          category     rating           city   latitude  longitude  \
1044   entire_home  4,77 (30)        Ushuaia -54.807290 -68.316100   
86     entire_home  4,55 (33)         Esquel -42.915770 -71.307150   
1071  private_room  4,85 (13)        Ushuaia -54.796494 -68.298038   
531    entire_home        NaN        Neuquén -38.953065 -68.065207   
851    entire_home  4,77 (26)        Ushuaia -54.808950 -68.319000   
527    entire_home        NaN        Neuquén -38.941620 -68.056090   
257    entire_home  4,84 (25)  Puerto Madryn -42.770730 -65.029470   
948    entire_home      Nuevo        Ushuaia -54.811890 -68.326950   
243    entire_home      Nuevo         Esquel -42.917491 -71.294685   
941    entire_home   4,88 (8)        Ushuaia -54.807163 -68.318520   

                                        characteristics    check_in  \
1044  ['2 camas', 'Monoambiente', '1 baño', 'Wifi', ...  2023-07-01   
86    ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-07-01   
1071  ['2 camas', '1 dormitorio', '1,5 baños compart...  2023-07-01   
531   ['No hay evaluaciones', '3 camas', '1 dormitor...  2023-07-02   
851   ['2 camas', 'Monoambiente', '1 baño', 'Wifi', ...  2023-07-01   
527   ['2 camas', 'Monoambiente', '0 baños', '1 sofá...  2023-07-01   
257   ['3 camas', '2 dormitorios', '2 baños', '1 cam...  2023-07-03   
948   ['2 camas', 'Monoambiente', '1 baño', 'Wifi', ...  2023-07-09   
243               ['4 camas', '1 dormitorio', '1 baño']  2023-07-01   
941   ['1 cama', '1 dormitorio', '1 baño', 'Wifi', '...  2023-07-04   

       check_out price_discounted price_original price_qualifier  \
1044  2023-07-06              NaN        $50 USD           noche   
86    2023-07-06              NaN        $20 USD           noche   
1071  2023-07-06              NaN        $26 USD           noche   
531   2023-07-08              NaN        $59 USD           noche   
851   2023-07-06          $59 USD        $68 USD           noche   
527   2023-07-06              NaN        $53 USD           noche   
257   2023-07-08              NaN       $114 USD           noche   
948   2023-07-14              NaN        $45 USD           noche   
243   2023-07-06          $52 USD        $65 USD           noche   
941   2023-07-09          $88 USD        $98 USD           noche   

      cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  \
1044             1000      2023-06-24       2023-05-01  239.919355   
86               1000      2023-06-24       2023-05-01  239.919355   
1071             1000      2023-06-24       2023-05-01  239.919355   
531               298      2023-06-24       2023-05-01  239.919355   
851              1000      2023-06-24       2023-05-01  239.919355   
527               298      2023-06-24       2023-05-01  239.919355   
257              1000      2023-06-24       2023-05-01  239.919355   
948              1000      2023-06-24       2023-05-01  239.919355   
243              1000      2023-06-24       2023-05-01  239.919355   
941              1000      2023-06-24       2023-05-01  239.919355   

             provincia  
1044  TIERRA DEL FUEGO  
86              C

In [41]:
# Vemos un ejemplo
df_alquileres[df_alquileres['longitude'] == -65.029470]

id_alojamiento                                               name  \
70         20364091                                Departamento Solmar   
257        53295849  Preciosa unidad de alquiler de 3 dormitorios e...   

        category      rating           city  latitude  longitude  \
70   entire_home  4,82 (137)  Puerto Madryn -42.76963  -65.02947   
257  entire_home   4,84 (25)  Puerto Madryn -42.77073  -65.02947   

                                       characteristics    check_in  \
70   ['2 camas', '1 dormitorio', '1 baño', '1 cama ...  2023-10-15   
257  ['3 camas', '2 dormitorios', '2 baños', '1 cam...  2023-07-03   

      check_out price_discounted price_original price_qualifier  \
70   2023-10-20              NaN        $47 USD           noche   
257  2023-07-08              NaN       $114 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna provincia  
70              1000      2023-06-24       2023-05-01  239.919355    CHUBUT  
257             1000      2023-06-24       2023-05-01  239.919355    CHUBUT

*ACCION:*
<br>
*Entendemos que la columna longitude, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columan 8: characteristics

In [42]:
# Verificamos los duplicados
df_alquileres['characteristics'].duplicated(keep="first").sum()

243

In [43]:
# Listamos algunos de los duplicados
df_alquileres[df_alquileres['characteristics'].duplicated(
    keep="first")].sample(10)

id_alojamiento                                              name  \
554            53256462  Cálida cabaña a orillas del lago con hidromasaje   
492  903093772979229006                  Dpto acogedor, céntrico y seguro   
398  821270097406761812                                       Pehuen 46 I   
588  793355312470293540                          Cabaña en los kilómetros   
556            12279136                       Cabaña de estilo patagónico   
130  838495226449697639                      Sofía. Céntrico departamento   
401  719485899637095838  Departamento amueblado moderno para dos personas   
28   890430561594058377                   Amplia casa con estacionamiento   
717            49546396         Departamento para 2. Excelente ubicación.   
555            53403878                            Peta Lodge Patagonia I   

        category      rating                     city   latitude  longitude  \
554  entire_home    5,0 (59)  San Carlos de Bariloche -41.104090 -71.437990   
492  entire_home       Nuevo               Cipolletti -38.941010 -67.980190   
398  entire_home         NaN                  Neuquén -38.954670 -68.046640   
588  entire_home   4,73 (11)  San Carlos de Bariloche -41.133160 -71.369120   
556  entire_home  4,76 (254)  San Carlos de Bariloche -41.134030 -71.353610   
130  entire_home     5,0 (3)            Puerto Madryn -42.769505 -65.033705   
401  entire_home         NaN                  Neuquén -38.973830 -68.059290   
28   entire_home       Nuevo                   Esquel -42.914878 -71.334675   
717  entire_home   4,89 (53)  San Carlos de Bariloche -41.134550 -71.299200   
555  entire_home   4,95 (93)  San Carlos de Bariloche -41.117900 -71.382700   

                                       characteristics    check_in  \
554  ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-09-18   
492      ['3 camas', '1 dormitorio', '1 baño', 'Wifi']  2023-07-01   
398  ['No hay evaluaciones', '2 camas', '1 dormitor...  2023-07-04   
588  ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-07-01   
556  ['1 cama', '1 dormitorio', '1 baño', '1 cama d...  2023-07-02   
130  ['3 camas', '1 dormitorio', '1 baño', 'Calific...  2023-07-01   
401  ['No hay evaluaciones', '1 cama', '1 dormitori...  2023-07-01   
28               ['2 camas', '1 dormitorio', '1 baño']  2023-07-01   
717  ['1 cama', 'Monoambiente', '1 baño', '1 cama q...  2023-07-07   
555  ['2 camas', '1 dormitorio', '1 baño', 'Estacio...  2023-08-17   

      check_out price_discounted price_original price_qualifier  \
554  2023-09-23              NaN       $150 USD           noche   
492  2023-07-06              NaN        $39 USD           noche   
398  2023-07-11              NaN        $38 USD           noche   
588  2023-07-06              NaN        $64 USD           noche   
556  2023-07-07              NaN        $55 USD           noche   
130  2023-07-06              NaN        $52 USD           noche   
401  2023-07-06              NaN        $60 USD           noche   
28   2023-07-06              NaN        $25 USD           noche   
717  2023-07-13              NaN        $70 USD           noche   
555  2023-08-23              NaN       $138 USD           noche   

     cant_resultados fecha_scrapping fecha_cambio_bna  cambio_bna  provincia  
554             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
492              298      2023-06-24       2023-05-01  239.919355    NEUQUEN  
398              298      2023-06-24       2023-05-01  239.919355    NEUQUEN  
588             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
556             1000      2023-06-24       2023-05-01  239.919355  RIO NEGRO  
130             1000      2023-06-24       2023-05-01  239.919355     CHUBUT  
401              298      2023-06-24       2023-05-01  239.919355    NEUQUEN  
28              1000      2023-06-24       2023-05-01  239.919355     CHUBUT  
717             1000      2023-06-24       2023-05-01  239.919355  RIO 

*ACCION:*
<br>
*Entendemos que la columna characteristics, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 9: check_in

In [44]:
# Verificamos la cantidad de duplicados por esta columna
df_alquileres['check_in'].duplicated(keep="first").sum()

920

*ACCION:*
<br>
*Entendemos que la columna check_in, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 10: check_out

In [45]:
# Verificamos la cantidad de duplicados por esta columna
df_alquileres['check_out'].duplicated(keep="first").sum()

916

*ACCION:*
<br>
*Entendemos que la columna check_out, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 11: price_discounted

In [46]:
# Verificamos la cantidad de duplicados por esta columna
df_alquileres['price_discounted'].duplicated(keep="first").sum()

938

*ACCION:*
<br>
*Entendemos que la columna price_discounted, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 12: price_original

In [47]:
# Verificamos la cantidad de duplicados por esta columna
df_alquileres['price_original'].duplicated(keep="first").sum()

868

*ACCION:*
<br>
*Entendemos que la columna price_original, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 13: price_qualifier

In [48]:
# Verificamos la cantidad de duplicados por esta columna
df_alquileres['price_qualifier'].duplicated(keep="first").sum()

1027

*ACCION:*
<br>
*Entendemos que la columna price_qualifier, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 14: cant_resultados

In [49]:
# Verificamos los duplicados
df_alquileres['cant_resultados'].duplicated(keep="first").sum()

1026

*ACCION:*
<br>
*Entendemos que la columna cant_resultados, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

Columna 15: fecha_scrapping

In [50]:
# Verificamos los duplicados
df_alquileres['fecha_scrapping'].duplicated(keep="first").sum()

1027

*ACCION:*
<br>
*Entendemos que la columna fecha_scrapping, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 16: fecha_cambio_bna

In [51]:
# Verificamos los duplicados
df_alquileres['fecha_cambio_bna'].duplicated(keep="first").sum()

1027

*ACCION:*
<br>
*Entendemos que la columna fecha_cambio_bna, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 17: cambio_bna

In [52]:
# Verificamos los duplicados
df_alquileres['cambio_bna'].duplicated(keep="first").sum()

1027

*ACCION:*
<br>
*Entendemos que la columna cambio_bna, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

+ Columna 18: provincia

In [53]:
# Verificamos los duplicados
df_alquileres['provincia'].duplicated(keep="first").sum()

1024

*ACCION:*
<br>
*Entendemos que la columna provincia, puede tener valores iguales para las diferentes observaciones*
<br>
*Por lo cual, no eliminamos segun los repetidos basados en esta columna*

*OBSERVACION GENERAL*
<br>
*En general entendemos que la mayoria de las columnas podrian tener valores repetidos evaluandolas individualmente.*
<br>
*Pero si consideramos que la variable id_alojamiento identifica de manera biunivoca un alojamiento.*
<br>
*Podemos decir que no es lo mismo, un duplicado en la columna id_alojamiento(que no deberia tener valores repetidos), que un duplicado en la columna price_original,*
<br>
*(que podria perfectamente tener valores repetidos en la seria).*


**<h1>3. Análisis de datos faltantes y limpieza</h1>**

+ Mostrar el porcentaje de datos faltantes de cada columna

In [82]:
# Verificacion inicial del dataset
df_alquileres.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1028 entries, 0 to 1079
Data columns (total 18 columns):
 #   Column            Non-Null Count  Dtype   
---  ------            --------------  -----   
 0   id_alojamiento    1028 non-null   int64   
 1   name              1028 non-null   object  
 2   category          1028 non-null   category
 3   rating            842 non-null    object  
 4   city              1024 non-null   object  
 5   latitude          1028 non-null   float64 
 6   longitude         1028 non-null   float64 
 7   characteristics   1028 non-null   object  
 8   check_in          1028 non-null   object  
 9   check_out         1028 non-null   object  
 10  price_discounted  191 non-null    object  
 11  price_original    1028 non-null   object  
 12  price_qualifier   1028 non-null   object  
 13  cant_resultados   1028 non-null   int64   
 14  fecha_scrapping   1028 non-null   object  
 15  fecha_cambio_bna  1028 non-null   object  
 16  cambio_bna        1028 non-nu

*En primera instancia observamos que solo nos quedan dos columnas con datos faltantes: rating y price_discounted*
<br>
*Para rating, tenenmos un 81.90 % de faltantes (842 de 1028).*
<br>
*Para rating, tenenmos un 18.58 % de faltantes (191 de 1028).*


+ ¿Cuáles son las 3 columnas con más datos faltantes?

*En nuestra observacion inicial, en principio tenemos dos columnas con datos faltantes solamente: rating y price_discounted*

**LIMPIEZA DE DE DATOS POR COLUMNAS**

In [56]:
df_alquileres.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1028 entries, 0 to 1079
Data columns (total 18 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   id_alojamiento    1028 non-null   int64  
 1   name              1028 non-null   object 
 2   category          1028 non-null   object 
 3   rating            842 non-null    object 
 4   city              1024 non-null   object 
 5   latitude          1028 non-null   float64
 6   longitude         1028 non-null   float64
 7   characteristics   1028 non-null   object 
 8   check_in          1028 non-null   object 
 9   check_out         1028 non-null   object 
 10  price_discounted  191 non-null    object 
 11  price_original    1028 non-null   object 
 12  price_qualifier   1028 non-null   object 
 13  cant_resultados   1028 non-null   int64  
 14  fecha_scrapping   1028 non-null   object 
 15  fecha_cambio_bna  1028 non-null   object 
 16  cambio_bna        1028 non-null   float64
 17  

+ Columna 0: id_alojamiento

In [61]:
df_alquileres['id_alojamiento'].nunique()

1028

In [63]:
df_alquileres['id_alojamiento'].isna().sum()

0

+ Columna 1: name

In [69]:
sorted(df_alquileres['name'].unique())

['"Alegria" vistas de ensueño en un lugar para enamorarse',
 '"Bathroom with LakeView Place and Retreat"',
 '"Casita" a estrenar, 21 m2 de luz',
 '"Cerros Nevados" departamento tipo loft.',
 '"Cipresales"     Cabaña  Villa la Angostura  Nqn',
 '"Doña Cota" - Amueblados Comahue',
 '"El Rancho" Sentirte en casa',
 '"La Encantada" en Villa los Coihues',
 '"La Posadita" Casa de Montaña, Miralejos',
 '"La casa de Cielo "',
 '"Los Amancays "\nCabaña de montaña',
 '"Tiny House de Ensueño"',
 '"Ty Bychan" \nLa Casita. Entre raíces galesas.',
 '0 - Pura Tranquilidad en el Corazón de Neuquén',
 '1- Pura Tranquilidad en el Corazón de Neuquén',
 '11-E Espectacular departamento Alto Neuquen',
 '2 - Pura tranquilidad en el corazón de Neuquén',
 '2 Amigos de viaje en un lugar comodo',
 '2 Habitaciones. 4 Personas en Casa con Pileta',
 '2 Personas. Lugar súper cómodo',
 '2 ambientes en el centro',
 '2x1 Central Family',
 '3 dormitorios. cama matrimonial + 3 de una plaza',
 '3- Pura tranquilidad en el 

+ Columna 2: category

In [79]:
# Vemos que tiene valores bien definidos, que no necesitan normalizar
df_alquileres['category'].unique()

array(['entire_home', 'private_room', 'shared_room'], dtype=object)

In [80]:
# Verificamos que no tienen  na
df_alquileres['category'].isna().sum()

0

In [81]:
# Cambiamos el tipo de la serie
df_alquileres['category'] = df_alquileres['category'].astype('category')